lib/new_relic/agent/instrumentation/aws_sqs/instrumentation.rb

```diff
@@ -0,0 +1,89 @@
+# This file is distributed under New Relic's license terms.
+# See https://github.com/newrelic/newrelic-ruby-agent/blob/main/LICENSE for complete details.
+# frozen_string_literal: true
+
+module NewRelic::Agent::Instrumentation
+  module AwsSqs
+    MESSAGING_LIBRARY = 'SQS'
+
+    def send_message_with_new_relic(*args)
```

Comment:
Outside of the `:action` and the error message, the three main methods in this module (`send_message_with_new_relic`, `send_message_batch_with_new_relic` and `receive_message_batch_with_new_relic`) are all looking the same to me. What do you think about breaking the shared code into a separate method?

[The Redis instrumentation takes this approach.](https://github.com/newrelic/newrelic-ruby-agent/blob/89119f566e03487c5eab3e49e9e7211720f166c5/lib/new_relic/agent/instrumentation/redis/instrumentation.rb#L47)

Reply:
cool, i did that here https://github.com/newrelic/newrelic-ruby-agent/pull/2679/commits/df6d3ca0acb102e7b55e2c0b89ed32c4a46aabdc